go/ql/lib/change-notes/2025-03-18-loggercall-type-format-specifier.md

```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* `LoggerCall::getAMessageComponent` no longer returns arguments to logger calls which correspond to the verb `%T` in a format specifier. This will remove false positives in "Log entries created from user input" (`go/log-injection`) and "Clear-text logging of sensitive information" (`go/clear-text-logging`), and it may lead to more results in "Use of constant `state` value in OAuth 2.0 URL" (`go/constant-oauth2-state`).
```

Comment:
Mirroring a review comment I think I wrote on another PR before: but this change note seems very implementation-oriented. Is that relevant to the target audience for this? I think if the goal is to address the FP in `go/log-injection`, then the change note should emphasise that. Otherwise, if we are communicating a change in the library, then we should consider whether the change has to be backwards compatible.

Reply:
I'm in two minds about whether to describe it as a change to the library, which happens to affect two queries in a very direct way, or as a change to two queries, with few/no details for anyone who might be using the predicate in the library that has been changed.

Reply:
I think that just reinforces my view in https://github.com/github/codeql/pull/19053#discussion_r2001390163 that this is a query change which is wrongly executed as a library change. If the intent is for this to fix the FP in the queries, then we should make sure that it is implemented and documented accordingly.